sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/util/Transport.java

```diff
@@ -105,6 +105,7 @@ private static ApiComponents apiComponentsFromUrl(String urlString) {
       ApiComponents components = apiComponentsFromUrl(servicePath);
       storageBuilder.setRootUrl(components.rootUrl);
       storageBuilder.setServicePath(components.servicePath);
+      storageBuilder.setBatchPath("batch" + components.servicePath);
```

Comment:
Evaluate using some sort of path joiner for this. We're taking advantage of the underlying knowledge that components.servicePath starts with a slash -- we shouldn't have to care about that.

Reply:
Thanks for pointing this out. Made the suggested change.